### ISSUE TITLE
Extremely poor validation results on self-trained checkpoints

### ISSUE THREAD

**Issue by Jessy-Huang** [tonyzhaozh/act#18]
Dear Tony.
Thank you for your excellent work on [ALOHA](https://github.com/tonyzhaozh/act?tab=readme-ov-file), I have tried to reproduce your work in the Mujoco simulation environment, and based on your [open source data](https://drive.google.com/drive/folders/1gPR03v05S1xiInoVJn7G7VJ9pDCnxq9O), The success rate should be around 90% for transfer cube, and around 50% for insertion.
I have trained and validated on your open source dataset, and I get results of around 54% for transfer cube, and around 14% for insertion.The results are very dismal, and I have not changed any of the parameter settings, and I would like to know what are some of the reasons for such a problem, or if I need to optimise in any way. The exact data can be viewed in the table below

Here are my training parameter settings
```shell
python3 imitate_episodes.py \
--task_name sim_transfer_cube_scripted \
--ckpt_dir ~/data/aloha/act/sim_transfer_cube_scripted/ckpt/ \
--policy_class ACT --kl_weight 10 --chunk_size 100 --hidden_dim 512 --batch_size 8 --dim_feedforward 3200 \
--num_epochs 2000  --lr 1e-5 \
--seed 0
```

Here are my eval parameter settings
```shell
python3 imitate_episodes.py \
--task_name sim_transfer_cube_scripted \
--ckpt_dir ~/data/aloha/act/sim_transfer_cube_scripted/ckpt/ \
--policy_class ACT --kl_weight 10 --chunk_size 100 --hidden_dim 512 --batch_size 8 --dim_feedforward 3200 \
--num_epochs 2000  --lr 1e-5 \
--seed 0 \
--eval
```
 The dataset obtained from the training is shown in the following link (Lark document, you need to register yourself, please note if you apply for permission to view)
https://iklxo6z9yv.feishu.cn/sheets/LRwosV4jnh7xokt8F8UcMT7snzh?from=from_copylink

**Comment by tonyzhaozh** [comment]
I suspect the mujoco version is messing things up. I updated the requirements yesterday: https://github.com/tonyzhaozh/act/commit/742c753c0d4a5d87076c8f69e5628c79a8cc5488

Could you try reinstalling these packages and evaluate the same checkpoints again? You would not need to retrain the policy.

**Comment by Jessy-Huang** [comment]
> I suspect the mujoco version is messing things up. I updated the requirements yesterday: [742c753](https://github.com/tonyzhaozh/act/commit/742c753c0d4a5d87076c8f69e5628c79a8cc5488)
> 
> Could you try reinstalling these packages and evaluate the same checkpoints again? You would not need to retrain the policy.

I have changed the version of mujuco and dm-control according to your instructions, their version information is as follows
```shell
(aloha) ➜  act-main conda list | grep dm-control
dm-control                1.0.14                   pypi_0    pypi
(aloha) ➜  act-main conda list | grep mujoco    
mujoco                    2.3.7                    pypi_0    pypi
(aloha) ➜  act-main 
```
The parameters used for model inference are as follows
```shell
python3 imitate_episodes.py \
--task_name sim_transfer_cube_scripted \
--ckpt_dir ~/data/aloha/act/sim_transfer_cube_scripted/ckpt/ \
--policy_class ACT --kl_weight 10 --chunk_size 100 --hidden_dim 512 --batch_size 8 --dim_feedforward 3200 \
--num_epochs 2000  --lr 1e-5 \
--seed 0 \
--eval
```
The result of this reasoning is as follows
```shell
Success rate: 0.54
Average return: 318.9

Reward >= 0: 50/50 = 100.0%
Reward >= 1: 40/50 = 80.0%
Reward >= 2: 35/50 = 70.0%
Reward >= 3: 27/50 = 54.0%
Reward >= 4: 27/50 = 54.0%

policy_best.ckpt: success_rate=0.54 avg_return=318.9

```
Does it need to be retrained or are there any other possibilities that led to a bad result？

In addition, by observing all the failed videos, I found that all the failures were caused by the right robotic arm failing to catch the square when grasping it leading to the failure of the final exchange of the square, is it necessary to introduce a model of object detection during grasping to ensure that the robotic arm can accurately grasp the object? I've put some of the failed gripping videos in the link below.

https://github.com/tonyzhaozh/act/assets/66520966/bff346ed-603f-44d4-980c-1c60a3de941b


https://github.com/tonyzhaozh/act/assets/66520966/55b63c30-8f51-491b-b427-9cebc382c065

**Comment by Jessy-Huang** [comment]
https://github.com/tonyzhaozh/act/assets/66520966/6348828e-9eec-4d39-8131-ea311083343c


https://github.com/tonyzhaozh/act/assets/66520966/1dc9a84f-f7ef-4db1-b36f-a74dac834c7f

**Comment by Jessy-Huang** [comment]
Dear Tony,
I followed your tips and updated mujoco and dm-control, but the results did not improve, see github issue for details


Best wishes,
Jessy











在 2024-01-30 03:13:38，"Tony Z. Zhao" ***@***.***> 写道：

I suspect the mujoco version is messing things up. I updated the requirements yesterday: 742c753

Could you try reinstalling these packages and evaluate the same checkpoints again? You would not need to retrain the policy.

**Comment by z-yf17** [comment]
> Dear Tony. Thank you for your excellent work on [ALOHA](https://github.com/tonyzhaozh/act?tab=readme-ov-file), I have tried to reproduce your work in the Mujoco simulation environment, and based on your [open source data](https://drive.google.com/drive/folders/1gPR03v05S1xiInoVJn7G7VJ9pDCnxq9O), The success rate should be around 90% for transfer cube, and around 50% for insertion. I have trained and validated on your open source dataset, and I get results of around 54% for transfer cube, and around 14% for insertion.The results are very dismal, and I have not changed any of the parameter settings, and I would like to know what are some of the reasons for such a problem, or if I need to optimise in any way. The exact data can be viewed in the table below
> 
> Here are my training parameter settings
> 
> ```shell
> python3 imitate_episodes.py \
> --task_name sim_transfer_cube_scripted \
> --ckpt_dir ~/data/aloha/act/sim_transfer_cube_scripted/ckpt/ \
> --policy_class ACT --kl_weight 10 --chunk_size 100 --hidden_dim 512 --batch_size 8 --dim_feedforward 3200 \
> --num_epochs 2000  --lr 1e-5 \
> --seed 0
> ```
> 
> Here are my eval parameter settings
> 
> ```shell
> python3 imitate_episodes.py \
> --task_name sim_transfer_cube_scripted \
> --ckpt_dir ~/data/aloha/act/sim_transfer_cube_scripted/ckpt/ \
> --policy_class ACT --kl_weight 10 --chunk_size 100 --hidden_dim 512 --batch_size 8 --dim_feedforward 3200 \
> --num_epochs 2000  --lr 1e-5 \
> --seed 0 \
> --eval
> ```
> 
> The dataset obtained from the training is shown in the following link (Lark document, you need to register yourself, please note if you apply for permission to view) https://iklxo6z9yv.feishu.cn/sheets/LRwosV4jnh7xokt8F8UcMT7snzh?from=from_copylink

I have the same prolem, have you solved it? Thanks.

**Comment by andrewcwlee** [comment]
@Jessy-Huang @z-yf17 Have you solved the problem?

**Comment by Jessy-Huang** [comment]
yes



---- Replied Message ----
| From | Andrew ***@***.***> |
| Date | 03/08/2024 05:33 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [tonyzhaozh/act] Extremely poor validation results on self-trained checkpoints (Issue #18) |

@***@***.*** Have you solved the problem?

**Comment by andrewcwlee** [comment]
I am experiencing a similar problem where the success rates are too low. How did you solve the problem? Any advice would be appreciated. Thank you!

**Comment by Jessy-Huang** [comment]
you can try to give a bigger num_epoch first



---- Replied Message ----
| From | Andrew ***@***.***> |
| Date | 03/08/2024 15:04 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [tonyzhaozh/act] Extremely poor validation results on self-trained checkpoints (Issue #18) |

I am experiencing a similar problem where the success rates are too low. How did you solve the problem? Any advice would be appreciated. Thank you!

**Comment by QinDeyun** [comment]
> 我也遇到类似的问题，成功率太低。你是如何解决这个问题的？任何意见，将不胜感激。谢谢！

You can try the following parameters, in my case the success rate is 100%.
 python imitate_episodes.py --task_name sim_transfer_cube_scripted --ckpt_dir ckpt_dir_batchsize_16_epoch_4000_TG --policy_class ACT --kl_weight 10 --chunk_size 100 --hidden_dim 512 --batch_size 16 --dim_feedforward 3200 --num_epochs 4000  --lr 2e-5 --seed 0 --eval
